project-addons/custom_account/account.py

```diff
@@ -118,6 +145,32 @@ class AccountInvoice(models.Model):
 
     date_due = fields.Date(string='Due Date',
                            readonly=True, states={'draft': [('readonly', False)], 'open': [('readonly', False)]})
+    state_web = fields.Char('State web', compute='_get_state_web', store=True)
+
+    @api.multi
+    @api.depends('state', 'payment_mode_id', 'payment_ids')
+    def _get_state_web(self):
+        for invoice in self:
+            val = invoice.state
+            if invoice.payment_mode_id:
+                if invoice.state == 'open' and invoice.returned_payment:
+                    self.state_web = 'returned'
+                elif invoice.state == 'paid':
+                    for payment in invoice.payment_ids:
+                        for payment_account in payment.move_id.line_id:
+                            if payment_account.account_id.code == '43120000' \
```

Comment:
Esta cuenta es mejor que la cojas de la transfer account del modo de pago, así haces que sea más genérica y te cubre casos futuros. Y sólo debería entrar en este for si tiene transfer account ya que sino no será recibo domiciliado y siempre sería paid.